### ISSUE TITLE
TrainResult doesn't log to tensorboard by default

### ISSUE THREAD

**Issue by xiadingZ** [PyTorchLightning/pytorch-lightning#3012]
This is my code:
```
        result = pl.TrainResult(minimize=loss)
        result.log('train_loss', loss, prog_bar=True)
```
tensorboard logger doesn't show `train_loss`.  
`EvalResult` is normal, it can log to tensorboard by default.

**Comment by justusschock** [comment]
cc @williamFalcon

**Comment by romesco** [comment]
Hey @xiadingZ, which version of `tensorboard` are you using? I ran into some issues if I wasn't using the PyPI version of `tensorboard==2.2.0`.

**Comment by xiadingZ** [comment]
2.2.0.  But I tried 2.3.0 and it have same issue, no TrainResult's log

**Comment by williamFalcon** [comment]
can you post a colab that replicates this?
i suspect you might be doing something weird for config since we test that things are actually logged when expected...

**Comment by teddykoker** [comment]
I have manually verified this works, we this colab: https://colab.research.google.com/drive/1aD1sEYNBLHISvnsUhRzyyOLZtMkk-LWs?usp=sharing.

Please send a colab or code so we can see what the issue is.

**Comment by stefanonardo** [comment]
I have the same issue: https://colab.research.google.com/drive/1s0kUXwXzO9t9Z1MXETohG7pu4dFJ6nej?usp=sharing

**Comment by ozen** [comment]
I have the same issue with tensorboard==2.2.0, pytorch-lightning==0.9.0.

Only the loss is not logged, other train metrics are logged.
